javatests/arcs/schematests/Arcs.kt

```diff
@@ -0,0 +1,185 @@
+        private val lifecycle = LifecycleRegistry(this)
+        override fun getLifecycle() = lifecycle
+    }
+
+
+    val arcHost = ArcHost(
+        context,
+        fakeLifecycleOwner.lifecycle,
+        schedulerProvider,
+        connectionFactory
+    )
+
+    private lateinit var allocator: Allocator
+
+    /**
+     * Start the arc containing the particles that provide the persistence. This should be run
+     * one time per instance, via [startIfNot].
+     */
+    private suspend fun startArc() {
+        /** This should probably live in an initializer somewhere. */
+        val hostRegistry = ExplicitHostRegistry().apply {
+            registerHost(arcHost)
+        }
+
+        allocator = Allocator.create(
+            hostRegistry,
+            EntityHandleManager(
+                arcId = "allocator",
+                hostId = "allocator",
+                time = JvmTime,
+                scheduler = schedulerProvider.invoke("allocator")
+            )
+        )
+
+        allocator.startArcForPlan("", WriteRecipePlan)
+    }
+
+    suspend fun stop() {
+        allocator.stopArc("!:testArc".toArcId())
+    }
+
+    private var started = false
+    val mutex = Mutex()
+
+    private suspend fun startIfNot() = mutex.withLock {
+        if (!started) {
+            startArc()
+            started = true
+        }
+    }
+
+    fun all0(): List<Level0> {
+        return runBlocking {
+            startIfNot()
+            arcHost.reader0.initialize().read()
+        }
+    }
+
+    fun put0(item: Level0) {
+        runBlocking {
+            startIfNot()
+            arcHost.writer0.initialize().write(item)
+        }
+    }
+
+    fun all1(): List<Level1> {
+        return runBlocking {
+            startIfNot()
+            arcHost.reader1.initialize().read()
+        }
+    }
+
+    fun put1(item: Level1) {
+        runBlocking {
+            startIfNot()
+            arcHost.writer1.initialize().write(item)
+        }
+    }
+
+    fun all2(): List<Level2> {
+        return runBlocking {
+            startIfNot()
+            arcHost.reader2.initialize().read()
+        }
+    }
+
+    fun put2(item: Level2) {
+        runBlocking {
+            startIfNot()
+            arcHost.writer2.initialize().write(item)
+        }
+    }
+}
+
+    /**
+     * This [ArcHost] is the home of the three particles used by the ParticipantPersistence recipe.
+     *
+     * It exposes their public methods to provide required read/write functionality.
+     */
+    class ArcHost(
+        context: Context,
+        lifecycle: Lifecycle,
+        schedulerProvider: SchedulerProvider,
+        connectionFactory: ConnectionFactory? = null
+    ) : AndroidHost(
+        context,
+        lifecycle,
+        schedulerProvider,
+        ::Reader0.toRegistration(),
+        ::Writer0.toRegistration(),
+        ::Reader1.toRegistration(),
+        ::Writer1.toRegistration(),
+        ::Reader2.toRegistration(),
+        ::Writer2.toRegistration()
```

Comment:
Tried switching to scanForParticles, but something wasn't working right. Will try again.

Reply:
Scan for particles requires all particles to use arcs_kt_particle blaze rule AND have an @TargetHost annotation. That's because we're not using annotation processors.

Reply:
Ah, thanks, I didn't know about that rule! I will fix this up in a followup PR.